src/components/KRadio/KRadio.vue

```diff
@@ -10,106 +10,196 @@
     },
   },
-  emits: ['change', 'update:modelValue'],
-  setup(props, { slots, emit, attrs }) {
-    const hasLabel = computed((): boolean => !!(props.label || slots.default))
+})
 
-    const isSelected = computed((): boolean => props.selectedValue === props.modelValue)
+const slots = useSlots()
 
-    const handleClick = (): void => {
-      emit('change', props.selectedValue)
-      emit('update:modelValue', props.selectedValue)
-    }
+const hasLabel = computed((): boolean => !!(props.label || slots.default))
 
-    const modifiedAttrs = computed(() => {
-      const $attrs = { ...attrs }
+const isSelected = computed(
+  (): boolean => props.selectedValue === props.modelValue,
+)
 
-      // delete classes because we bind them to the parent
-      delete $attrs.class
+const emit = defineEmits(['change', 'update:modelValue'])
 
-      return $attrs
-    })
+const handleClick = (): void => {
+  emit('change', props.selectedValue)
+  emit('update:modelValue', props.selectedValue)
+}
 
-    return {
-      hasLabel,
-      isSelected,
-      modifiedAttrs,
-      handleClick,
-    }
-  },
+const attrs = useAttrs()
+
+const modifiedAttrs = computed(() => {
+  const $attrs = { ...attrs }
+
+  // delete classes because we bind them to the parent
+  delete $attrs.class
+
+  return $attrs
 })
 </script>
 
 <style lang="scss" scoped>
 @import '@/styles/variables';
 @import '@/styles/functions';
 
+$text-color-default: var(--black-45, rgba(0, 0, 0, 0.45));
+$text-color-card: color(black-500);
+$border-color-card: color(grey-300);
+$background-color-card-checked: color(blue-100);
+$border-color-card-checked: color(blue-300);
+$background-color-card-disabled: color(grey-200);
+
 .k-radio-label {
   font-size: var(--type-sm, type(sm));
 }
 
 .k-radio-description {
   padding-top: var(--spacing-xxs);
-  padding-left: var(--spacing-lg);
   font-size: var(--type-sm, type(sm));
   line-height: 20px;
-  color: var(--black-45, rgba(0, 0, 0, 0.45));
+  color: $text-color-default;
 }
 
-.k-radio-label:has(+ .k-radio-description) {
-  font-weight: 600;
+// default radio input styling
+.k-radio-default {
+  .k-radio-description {
+    padding-left: var(--spacing-lg);
+  }
+
+  .k-radio-label:has(+ .k-radio-description) {
+    font-weight: 600;
+  }
 }
 
+// card radio input styling
+.k-radio-card {
+  align-items: center;
+  background-color: color(white);
+  border: 1px solid $border-color-card;
+  border-radius: var(--spacing-xxs);
+  cursor: pointer;
+  display: flex;
+  padding: 16px;
+  flex-direction: column;
+
+  .k-input {
+    display: none;
+  }
+
+  .k-radio-label {
+    color: $text-color-card;
+    font-size: var(--type-sm, type(sm));
+    font-weight: 500;
+  }
+
+  .k-radio-content-before {
+    margin-bottom:  var(--spacing-sm);
+  }
+
+  .k-radio-content-after {
+    margin-top:  var(--spacing-sm);
+  }
+
+  &:has(.k-input:disabled) {
+    background-color: $background-color-card-disabled;
+    cursor: not-allowed;
+    opacity: 0.6;
```

Comment:
Yeah shouldn't be a problem even if the bug is there - since this is for `disabled` state only.
Regarding the `pointer-events: none;` - it isn't needed since it's handled by the input element when `disabled` attribute is passed to it.